packages/@ember/-internals/glimmer/lib/helper.ts

```diff
@@ -256,7 +265,11 @@ setHelperManager(() => SIMPLE_CLASSIC_HELPER_MANAGER, Wrapper.prototype);
   @public
   @since 1.13.0
 */
-export function helper(helperFn: HelperFunction): HelperFactory<SimpleHelper> {
+export function helper<
+  T = unknown,
+  P extends unknown[] = unknown[],
+  N extends Dict<unknown> = Dict<unknown>
```

Comment:
I *think* we want these not to have defaults—they should "just work" by way of the definition.

```suggestion
  T,
  P extends unknown[],
  N extends Dict<unknown>
```

(This is how the types on DefinitelyTyped work, and I came to believe that the default types for `ember-modifier` were the wrong move as well. That’s why `modifier` itself [does *not* set defaults](https://github.com/ember-modifier/ember-modifier/blob/709949c6ce33f652c96b6a70dcea0b543ee98967/addon/-private/functional/modifier.ts#L26-L32); and [the `FunctionalModifier` return type](https://github.com/ember-modifier/ember-modifier/blob/709949c6ce33f652c96b6a70dcea0b543ee98967/addon/-private/functional/modifier.ts#L5-L9) does only for backwards compatibility.)